LEGO1/lego/legoomni/src/worlds/hospital.cpp

```diff
@@ -580,16 +580,16 @@ MxBool Hospital::HandleControl(LegoControlManagerNotificationParam& p_param)
 		case HospitalScript::c_Door_Ctl:
 			DeleteObjects(&m_atomId, HospitalScript::c_hho002cl_RunAnim, HospitalScript::c_hho006cl_RunAnim);
 
-			if (m_unk0x100 == 1) {
-				m_hospitalState->m_unk0x08 = 15;
+			if (m_interactionMode == 1) {
+				m_hospitalState->m_state = HospitalState::e_exitViaInfo;
```

Comment:
```suggestion
				m_hospitalState->m_state = HospitalState::e_exitViaDoor;
```

The current code is (imho) correct, but not a direct translation from the previous code (see above).